src/stirling/binaries/go_binary_parse_profiling.cc

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2018- The Pixie Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "src/common/base/base.h"
+#include "src/common/base/env.h"
+#include "src/stirling/source_connectors/socket_tracer/uprobe_symaddrs.h"
+
+using px::StatusOr;
+using px::stirling::PopulateGoTLSDebugSymbols;
+using px::stirling::obj_tools::DwarfReader;
+using px::stirling::obj_tools::ElfReader;
+
+//-----------------------------------------------------------------------------
+// Main
+//-----------------------------------------------------------------------------
```

Comment:
leave a comment on the purpose of the utility. There's an explanation on the PR, but would be useful to encode an explanation into this file.